pkg/vminetworkscontroller/vmi_controller_test.go

```diff
@@ -299,27 +306,29 @@ var _ = Describe("vmi IPAM controller", Serial, func() {
 	)
 })
 
-func dummyVM(nadName string) *virtv1.VirtualMachine {
+type VMISpecFunc func(nadName string) virtv1.VirtualMachineInstanceSpec
+
+func dummyVM(nadName string, specFunc VMISpecFunc) *virtv1.VirtualMachine {
```

Comment:
so you're giving it the nad name, and a function whose input is the nad name as parameters ...

why not simply give it the calculated spec, and pass it from the outside ? IMO it's less convoluted.

Reply:
right done